Updating docs from 0.6.8 to 0.7

joshring opened the pull request:
Changelog:

- Language changes from: https://github.com/c3lang/c3c/blob/master/releasenotes.md
- Fixed the ordering of the sidebar being alphabetical instead of what we intended
- I avoided some [Stdlib changes](https://github.com/c3lang/c3c/blob/master/releasenotes.md#stdlib-changes) as I am not best placed to update them 

Each commit is generally addressing a specific point, I recommend looking at the commits individually to make this more digestible.

This PR is live here: https://experimental-site-testing-only.github.io/

joshring committed:
fixing ordering of items in the sidebar

joshring committed:
removed Discourse link
Removed macro &ref and $varef parameters.
mentioned pool in the "Getting Started" section: "- Automically free memory after use in `@pool` context."

joshring committed:
removing @adhoc
changing (< >) to {} and module_name<> to module_name{}

joshring committed:
def -> alias
def -> attrdef for attributes

joshring committed:
fault, faultdef, alias

joshring committed:
$foreach

joshring committed:
$for

joshring committed:
$switch

joshring committed:
catch excuse switch removed

joshring committed:
@return!

joshring committed:
[?]

joshring committed:
distinct

joshring committed:
`temp_new`, `temp_alloc`, `temp_new_array`, `temp_alloc_array`

joshring committed:
switch update to spec from discord

joshring committed:
bump the version

lerno commented on src/content/docs/Generic Programming/generics.md:
````diff
@@ -13,27 +13,27 @@ For generic modules, the generic parameters follows the module name:
 ```c3
 // TypeA, TypeB, TypeC are generic parameters.
-module vector(<TypeA, TypeB, TypeC>);
+module vector{ TypeA, TypeB, TypeC };
````
I found `module vector {TypeA, TypeB, TypeC}` is more natural. Use that instead.

lerno commented on src/content/docs/Generic Programming/generics.md:
```diff
@@ -13,27 +13,27 @@ For generic modules, the generic parameters follows the module name:
-// module custom_type(<Type, VALUE>)  
-module custom_type(<float, 3>);
+// module custom_type{ Type, VALUE }
+module custom_type{ float, 3 };
```
Similar here: `module custom_type {float, 3};`

lerno commented on src/content/docs/Generic Programming/generics.md:
```diff
@@ -44,18 +44,18 @@ Including a generic module works as usual:
 
-def FooFloat = Foo(<float, double>);
-def test_float = foo_test::test(<float, double>);
+alias FooFloat = Foo{ float, double };
```
For these we do the same `Foo {float, double};`

lerno commented on src/content/docs/Language Common/optionals-advanced.md:
````diff
@@ -168,49 +163,32 @@ fn void find_file_and_test()
 }
 ```
 
 ### Find empty Optional and switch on Excuse
````
This section should be completely removed

lerno commented on src/content/docs/Language Fundamentals/comments.md:
```diff
@@ -46,12 +46,12 @@ The following was extracted:
 |---------------|----------------------------------------------------:|
 | `@param     ` |            `@param [<ref>] <param> [<description>]` |
```
The `:` is now mandatory in front of the description, so it's `[ : <description>]`

lerno commented on src/content/docs/Language Overview/examples.md:
````diff
@@ -578,7 +575,7 @@ Read more about compile time execution [here](/generic-programming/compiletime/)
 
 ```c3
-module stack(<Type>);
+module stac{ Type };
````
Typo here, should be `module stack {Type}`

joshring committed:
undoing format change
switch needed to be added to fault handling
removing unneeded optional feature explanation
generics syntax tweak
added typedef to primer
